lib/msal-node/src/client/PublicClientApplication.ts

```diff
@@ -74,4 +84,51 @@ export class PublicClientApplication extends ClientApplication implements IPubli
             throw e;
         }
     }
+
+    /**
+     * Acquires a token by requesting an Authorization code then exchanging it for a token.
+     */
+    async acquireTokenInteractive(request: InteractiveRequest): Promise<AuthenticationResult> {
+        const { verifier, challenge } = await this.cryptoProvider.generatePkceCodes();
+        const { openBrowser, successTemplate, errorTemplate, ...remainingProperties } = request;
+
+        const loopbackClient = new LoopbackClient();
+        const authCodeListener = loopbackClient.listenForAuthCode(successTemplate, errorTemplate);
+        const port = await loopbackClient.getPort().catch((err) => {
+            // Close server and re-throw any error thrown when getting port
+            loopbackClient.closeServer();
+            throw err;
+        });
```

Comment:
Could the implementation detail of the port be hidden behind a `getRedirectUri()` function?

Reply:
Done